JitHub.AuthFunction/GithubAuth.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Octokit;
+using System.Text.Json;
+
+namespace JitHub.AuthFunction
+{
+    public static class GithubAuth
+    {
+        public static async Task<OauthToken> Detokenize(string code)
+        {
+
+            try
+            {
+                string clientId = Environment.GetEnvironmentVariable("JithubClientId");
+                string appSecret = Environment.GetEnvironmentVariable("JithubAppSecret");
+
+                var request = new OauthTokenRequest(clientId, appSecret, code);
+                var gitHubClient = new GitHubClient(new ProductHeaderValue("JitHub"));
+                var token = await gitHubClient.Oauth.CreateAccessToken(request);
+
+                return token;
+            }
+
+            catch
+            {
+                return null;
+            }
+
+        }
+
+        public static string ProcessRequest(HttpRequest req)
+        {
+            try
+            {
+                string temporaryCode = req.Query["tempCode"];
```

Comment:
is it planned to have the FE pass a param called `tempCode`?

Reply:
I was thinking tepmCode since github considers it a temporary code. We can change it to code to be more concise

Reply:
that's fine as long as FE and Function are consistent within themselves